File: springbootjparest/src/main/java/com/scarc/springbootjparest/models/Responsible.java
package com.scarc.springbootjparest.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Responsible extends User {
    @ManyToOne
    private Mail mail;

    @OneToMany(mappedBy = "responsible", orphanRemoval = true, cascade = CascadeType.ALL)
    private List<Participation> responsibleFor;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }
}

File: KOTLIN/Wirtschaftstag/Kotlin_Wirtschaftstag/androidApp/src/main/java/me/scarc/androidApp/data/Responsible.kt
package me.scarc.androidApp.data

data class Responsible(
    var id: Int,
    var name: String,
    var pwdToken: String,
    var email: String,
    var mail: Mail,
    var responsibleList: List<Participation>
) : User(id, name, pwdToken, email) {

}
